File: floyds.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/Floyd%27s_triangle

floyds = -> r,n=1 { (1...r).each {|r| (1..r).each { print n < 10 ? "  #{n}" : " #{n}"; n += 1 }; print "\n" } }

p floyds[14]

File: floyds.js
#!/usr/bin/env node
// floyds triangle

function floyds(rows) {
  var n = 1;
  for (var r = 1; r <= rows; r++) {
    for (var c = 1; c <= r; c++) {
      process.stdout.write(
        n >= 100 ? n+" " : n >= 10 ? " "+n+" " : "  "+n+" "
      );
      n++;
    };
    console.log("");
  };
  console.log("");
}

floyds(5);
floyds(14);